watchface

```diff
@@ -0,0 +1,259 @@
+    local dconfsetting="$1"
+    local filename="$2"
+    doWatchCommand "ceres" "dconf write ${dconfsetting} '\"file://${filename}\"'"
+}
+
+function pushFiles {
+    local user="$1"
+    local sourcedir="$2"
+    local destdir="$3"
+    if [ "$ADB" = true ] ; then
+        adb push ${sourcedir} "${destdir}"
+    else
+        scp -P"${WATCHPORT}" -r ${sourcedir} "${user}@${WATCHADDR}:${destdir}"
+    fi
+}
+
+function pullFiles {
+    local user="$1"
+    local sourcedir="$2"
+    local destdir="$3"
+    if [ "$ADB" = true ] ; then
+        adb pull ${sourcedir} "${destdir}"
+    else
+        scp -P"${WATCHPORT}" -r "${user}@${WATCHADDR}:${sourcedir}" "${destdir}"
+    fi
+}
+
+function pushWatchface {
+    pushFiles "root" "${1}"'/usr/share/*' "/usr/share/"
+}
+
+function pushWallpaper {
+    local source="$1"
+    local wallpaper="$(basename "$1")"
+    local destination="/usr/share/asteroid-launcher/wallpapers/full/${wallpaper}"
+    pushFiles "root" "${source}" "${destination}"
+    setDconf "/desktop/asteroid/background-filename" "${destination}"
+}
+
+function restartCeres {
+    doWatchCommand "root" "systemctl restart user@1000"
+}
+
+function activateWatchface {
+    setDconf "/desktop/asteroid/watchface" "/usr/share/asteroid-launcher/watchfaces/${1}.qml"
+}
+
+function setTimeZone {
+    doWatchCommand "root" "timedatectl set-timezone $1"
+}
+
+function forgetKeys {
+    ssh-keygen -R ${WATCHADDR}
+    ssh-keygen -R watch
+}
+
+function watchfaceClone {
+    local files
+    mapfile -t files <<<"$(find . -path './'"$1"'/*' -type f | awk '{ dst=$0 ; gsub(/'"$1"'/,"mmmm",dst);  print $0 " " dst }')"
+    for myfile in "${files[@]}" ; do
+        IFS=" " read -r -a onefile <<< "${myfile}"
+        #echo "Copying from ${onefile[0]} to ${onefile[1]}"
+        destdir="${onefile[1]%/*}"
+        extension="${onefile[1]##*.}"
+        mkdir -p "${destdir}"
+        cp "${onefile[0]}" "${onefile[1]}"
+        if [ "${extension}" = "qml" ] ; then
+            sed -i "s/$1/$2/g" "${onefile[1]}"
+        fi
+    done
+}
+
+function deployface {
+    echo "Deploying $1"
+    pushWatchface "$1"
+    activateWatchface "$1"
+    restartCeres
+}
+
+function testface {
+    echo "Testing $1"
+    qmlscene "$1" loader.qml
+}
+
+function cloneface {
+    local SOURCE="$1"
+    local DEST="$2"
+    echo "Cloning $1 into $2"
+    if [ ! -d "${SOURCE}" ] ; then
+        echo "Error:  ${SOURCE} does not exist, exiting program"
+        exit
+    fi
+    if [ -e "${DEST}" ] ; then
+        echo "Error:  ${DEST} already exists, exiting program"
+        exit
+    fi
+    watchfaceClone "${SOURCE}" "${DEST}"
+}
+
+# These are the defaults for SSH access
+WATCHPORT=22
+WATCHADDR=192.168.2.15
+# These are the defaults for local QEMU target
+QEMUPORT=2222
+QEMUADDR=localhost
+# Assume no ADB unless told otherwise
+ADB=false
+# get this host's IPv4 address
+MYADDR=$(hostname -I |grep -oP 192.168.2.[[:digit:]]+)
+# we're not done until we say we are
+DONE=false
```

Comment:
I think it would be clearer if named in a way similar to SKIP_INTERACTIVE_PROMPT (and with the logic flipped)

Reply:
Agreed.  Fixed!